roles/network_plugin/calico/tasks/install.yml

```diff
@@ -227,7 +227,7 @@
               "cidr": "{{ calico_pool_cidr | default(kube_pods_subnet) }}",
               "ipipMode": "{{ calico_ipip_mode }}",
               "vxlanMode": "{{ calico_vxlan_mode }}",
-              "natOutgoing": {{ nat_outgoing|default(false) and not peer_with_router|default(false) }}
+              "natOutgoing": {{ nat_outgoing|default(false) and ((not peer_with_router|default(false)) or (peer_with_router_nat_outgoing|default(false))) }}
```

Comment:
Can we control this use case with the existing `nat_outgoing` only?

Reply:
Sure, we can remove the mutual exclusivity between `nat_outgoing` and `not peer_with_router` by change the logic.

But it may causes for users that enable `peer_with_router` to get side-effect.

Regardless, the direction of concise and complete improvement seems to fit the content of the review, so I made a new commit.